src/status_im/utils/number.cljs

```diff
@@ -13,3 +16,48 @@
   (let [scale (Math/pow 10 decimal-places)]
     (/ (Math/round (* n scale))
        scale)))
+
+(defn- caught-invalid-number []
+  (throw (js/Error "Invalid Number")))
+
+(defn- with-precision [precision input-number]
+  (gstring/format (str "%." precision "f") input-number))
```

Comment:
Excellent work in this namespace @ibrkhalil!

I saw this in the documentation for [google.string.format](https://google.github.io/closure-library/api/goog.string.format.html). If you are in the mood, you could replace calls to `with-precision` to use `toFixed` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed).

> Performs sprintf-like conversion, i.e. puts the values in a template. DO NOT use it instead of built-in conversions in simple cases such as 'Cost: %.2f' as it would introduce unnecessary latency opposed to 'Cost: ' + cost.toFixed(2).

Reply:
Thanks Icaro
Is there any difference between the JS function toFixed and CLJS's with-precision .. if not, let's leave it as is.

Reply:
Thanks for the recommendation Icaro, Of course that's helpful to know
I'll add the case for millions and billions alongside their respective tests and will try 'toFixed' alongside